src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -492,6 +491,11 @@ bool containsModopt(INamedTypeSymbol symbol)
             }
         }
 
+        private static INamedTypeSymbol GetTupleUnderlyingTypeOrSelf(INamedTypeSymbol type)
```

Comment:
>private static INamedTypeSymbol GetTupleUnderlyingTypeOrSelf(INamedTypeSymbol type) [](start = 8, length = 83)

It feels like we should be able to share this helper between compilers by adding it to ITypeSymbolHelpers type.